src/hotspot/share/opto/parse1.cpp

```diff
@@ -2174,16 +2173,50 @@ PhiNode *Parse::ensure_memory_phi(int idx, bool nocreate) {
   return phi;
 }
 
+// Passive Materialization
+// ------------------------
+// C2 has to materialize a virtual object at the merging point because the object has been materialized in
+// any other predecessor. For instance:
+//
+// obj(virtual)     obj(materialized)
+//              \      /
+// Reion(,c1,c2) |    |
+//      \        |    |
+//       \c      |    |
+//   o3 = Phi   (o1, o2)
+//
+// Node* var is o1;
+// r[pnum] is c1;
+//
+// There are 2 cases:
+// 1) materialize m from map().
+// 2) materialize n from newin.
+// so from_map is either from current map or newin of merge_common().
+// for case 1, the control of from_map is Region, so we need to adjust it to c1.
```

Comment:
Having trouble understanding this control update. `c1` refers to control node of virtual object. Why would we update the control node of `kit`?

> the control of from_map is Region

Not sure what `Region` is here

Reply:
here Region is the control of the current map, which is associated with the merged block.